packages/investor-yearn/src/api/api.ts

```diff
@@ -0,0 +1,333 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { ChainAdapter } from '@shapeshiftoss/chain-adapters'
+import { ChainTypes } from '@shapeshiftoss/types'
+import { Vault, Yearn } from '@yfi/sdk'
+import { BigNumber } from 'bignumber.js'
+import { toLower } from 'lodash'
+import isNil from 'lodash/isNil'
+import Web3 from 'web3'
+import { TransactionReceipt } from 'web3-core/types'
+
+import { MAX_ALLOWANCE } from '../constants/allowance'
+import { erc20Abi } from '../constants/erc20-abi'
+import { ssRouterContractAddress } from '../constants/router-contract'
+import { ssRouterAbi } from '../constants/ss-router-abi'
+import { yv2VaultAbi } from '../constants/yv2Vaults-abi'
+import { bnOrZero } from '../utils/bignumber'
+import { buildTxToSign } from '../utils/buildTxToSign'
+import {
+  Allowanceinput,
+  ApproveEstimatedGasInput,
+  ApproveInput,
+  APYInput,
+  BalanceInput,
+  TokenInput,
+  TxEstimatedGasInput,
+  TxInput
+} from './yearn-types'
+
+export type ConstructorArgs = {
+  adapter: ChainAdapter<ChainTypes.Ethereum>
+  providerUrl: string
+}
+
+export type YearnVault = Vault
+
+export class YearnVaultApi {
+  public adapter: ChainAdapter<ChainTypes.Ethereum>
+  public provider: any
+  public jsonRpcProvider: JsonRpcProvider
+  public web3: Web3
+  public vaults: Vault[]
+  private yearnSdk: Yearn<1>
+  private ssRouterContract: any
+
+  constructor({ adapter, providerUrl }: ConstructorArgs) {
+    this.adapter = adapter
+    this.provider = new Web3.providers.HttpProvider(providerUrl)
+    this.jsonRpcProvider = new JsonRpcProvider(providerUrl)
+    this.web3 = new Web3(this.provider)
+    this.yearnSdk = new Yearn(1, { provider: this.jsonRpcProvider, disableAllowlist: true })
```

Comment:
please make network injectable here as we're spiking on testnets at the moment too

Reply:
Done.